src/reader-activation/index.js

```diff
@@ -397,9 +399,35 @@ function attachNewsletterFormListener() {
 		} );
 	};
 
-	// For third-party forms, set reader data on form submit. For first-party forms, listen for the custom event upon successful signup response.
-	document.querySelectorAll( thirdPartyForms.join( ',' ) ).forEach( el => attachHandler( el ) );
+	const gformIds = [];
+
+	// For third-party forms, set reader data on form submit. Also detect Gravity Forms.
+	document.querySelectorAll( thirdPartyForms.join( ',' ) ).forEach( el => {
+		// If the form id stats with gform_ and it has a data-formid prop, it's a gravity form.
+		const isGravityForm = el.id.startsWith( 'gform_' ) && el.hasAttribute( 'data-formid' );
+		if ( isGravityForm ) {
+			gformIds.push( parseInt( el.getAttribute( 'data-formid' ) ) );
+			return;
+		}
+		// If not a gravity form, just attach the handler.
+		attachHandler( el );
+	} );
+
+	// First-party forms success event listener.
 	document.querySelectorAll( newspackForms.join( ',' ) ).forEach( el => attachHandler( el, 'newspack-newsletters-subscribe-success' ) );
+
+	// Gravity forms handlers.
+	document.addEventListener( 'gform/post_render', event => {
+		if ( gformIds.includes( event.detail.formId ) ) {
+			if ( ! window.gform ) {
+				return;
+			}
```

Comment:
Nit: this is a bit more terse and also guards against unhandled errors in case of nonexistent nested properties:

```suggestion
		if ( window.gform?.utils?.addAsyncFilter && gformIds.includes( event.detail?.formId ) ) {
```